src/test/java/org/isf/medicalstock/test/Tests.java

```diff
@@ -1194,6 +1197,23 @@ public void testMovementGetOrigin() throws Exception {
 		assertThat(movement).isNotNull();
 		assertThat(movement.getSupplier()).isEqualTo(movement.getOrigin());
 	}
+	
+	@Test
+	public void testDeleteLastMovement() throws Exception {
+		int code = setupTestMovement(false);
+		Optional<Movement> movement = movementIoOperationRepository.findById(code); 
+		assertThat(movement).isPresent();
+		movBrowserManager.deleteLastMovement(movement.get());
+		Optional<Movement> movement2 = movementIoOperationRepository.findById(code); 
+		assertThat(movement2).isNotPresent();
+	}
+	
+	@Test
+    public void testDeleteLastMovementDenied() throws Exception {
+		int code = setupTestMovement(false);
+		Optional<Movement> movement = movementIoOperationRepository.findById(code - 1); 
+        assertThrows(NoSuchElementException.class, () -> movBrowserManager.deleteLastMovement(movement.get()));
```

Comment:
why you are testing the deletion of a non existing movement (code - 1) ?
You should try the deletion of an existing movmement that for some reason cannot be deleted anymore
Please check also the indentation